### ISSUE TITLE
Feature request: subcommand support

### ISSUE THREAD

**Issue by clawoflight** [dffischer/argparse-manpager#1]
This is really cool, but it only seems to use the primary parser.
I have 1-2 levels of subparsers in my ArgumentParser.

Could you add support for that?

**Comment by dffischer** [comment]
Yes, subparsers are currently ignored. I did not have the need for these up to now, but do agree that this would be a useful feature.

The main conceptual problem occurring to me is that manual pages only have two levels of hierarchy, while subparsers may nest arbitrarily deep. The first sensible structure coming to my mind is to flatten this out and place any subparser sections besides argument groups below _OPTIONS_. So this would be a first draft of where to end up:

```
NAME
SYNOPSIS
DESCRIPTION
OPTIONS
  positional arguments
  optional arguments
  argument group 3
  argument group 4
  ...
  subparser 1
  subparser 2
  subsubparser 2.1
  subsubparser 2.2
  subparser 3
  ...
SEE ALSO
AUTHORS
```

(Note that positional and optional arguments are implemented as argument groups by _argparse_ itself.)

So I guess adding support for subparsers would indeed be possible. But as it is often the case in open source development, I highly suspect it to be complex enough to exceed my currently available time contingent. However, I would like to take a look into your parser structure. This would provide me with a good use case to test and program against. Could you point me to your project?

P.S.: I wonder whether [the famous quote](http://refspace.com/quotes/Linus_Torvalds/Q503) also applies to levels of subparsers...

**Comment by clawoflight** [comment]
Thanks for the reply!

That draft looks good. Another option (if I understood the constraints correctly) would be to add a first-level Heading like _COMMANDS_ or similar and flatten the subparsers in there.

I fully respect that no one has unlimited time. Just wanted to put that on your radar and get an assessment :) I'm not that used to python yet, but I may be able to contribute, if that helps.

I can't link you the repo because the project leader wants to open-source it on the first useable release, which is soon. However, this is the [entry point](https://vps1.piater.name/commie/#zOiYcvq5) for the CLI.
You can run it if you remove the local imports and the actual functionality. Let me know if you can't follow the code.
We're making a light (as in LED light) alarm clock with a raspberry pi and an arduino.

I don't know, I feel like I am using them as intended. I also don't have more than 2 levels of nesting :)
